### ISSUE TITLE
Monitor grpc traffic via devtools

### ISSUE THREAD

**Issue by mehmetf** [flutter/devtools#2044]
Currently devtools can monitor HTTP traffic. Some transport implementations (in particular package:grpc) operate directly on dart:io Socket. The ideal goal of this would be:

- Monitor socket traffic.
- Be able to interpret and decode grpc traffic (protos) and show details about the payload in the tooling

@yohanln has mentioned this would be a good intern project.

**Comment by mehmetf** [comment]
Also @jacob314 and @bkonyi as FYI

**Comment by bkonyi** [comment]
@zichangg has actually already implemented socket profiling in dart:io, I just haven't gotten around to surfacing it in DevTools yet (I'll hopefully get around to it in Q3).

**Comment by zichangg** [comment]
Right! But my implementation is only monitoring statistics of socket(without looking at the content). It probably needs more work for grpc.

**Comment by thedejifab** [comment]
Is anyone working on this at the moment? Currently building stuff using GRPC and I need to monitor my traffic as I do with REST API calls but this is a blocker.

**Comment by zichangg** [comment]
We will start the work very soon!

**Comment by thedejifab** [comment]
Cool @zichangg. I think I'm actually interested in working on this

**Comment by kenzieschmoll** [comment]
There is an in-progress CL for this. @bkonyi is going to pick up on the existing work when he has some bandwidth.

**Comment by kenzieschmoll** [comment]
Some work for this effort was landed here: https://github.com/grpc/grpc-dart/pull/392, but we need to audit what work is left to be done to support debugging gRPC requests with DevTools

**Comment by EnesKaraosman** [comment]
Any update here?

**Comment by CoderDake** [comment]
Currently at the back of a long backlog.

**Comment by loeffel-io** [comment]
+1

**Comment by srawlins** [comment]
@kenzieschmoll: @bkonyi and I were thinking this might be better suited to be a DevTools extension. WDYT?
